modules/sdk-coin-ada/src/ada.ts

```diff
@@ -36,9 +43,27 @@ export interface SignTransactionOptions extends BaseSignTransactionOptions {
   prv: string;
 }
 
+interface RecoveryOptions {
+  userKey?: string; // Box A
+  backupKey?: string; // Box B
+  bitgoKey: string; // Box C
+  recoveryDestination: string;
+  krsProvider?: string;
+  walletPassphrase?: string;
+  startingScanIndex?: number;
+  scan?: number;
+}
+
+interface AdaTx {
+  serializedTx: string;
+  scanIndex: number;
+}
+
 export type AdaTransactionExplanation = TransactionExplanation;
 
 export class Ada extends BaseCoin {
+  private static readonly TestnetEnv: Set<EnvironmentName> = new Set(['mock', 'test']);
+  private static readonly MainnetEnv: Set<EnvironmentName> = new Set(['prod', 'staging']);
```

Comment:
there could be a better way to support this.
we use the convention where a coin has a `t` prefix for non-mainnet environments.
e.g.
```
ada vs tada
sol vs tsol
dot vs tdot
```

Reply:
Changed it to comparing whether the chain is 'tada' or 'ada' when determining which address type to use